### DISCUSSION TITLE
Not able to get the `width` of window

### DISCUSSION THREAD

**Discussion by dilawar** [slint-ui/slint#1229] [Q&A] [answered]
I am under the impression that I can get a property using `get_<property>`. Are `width` and `height` of a `Window` properties? Following doesn't compile.

```
use slint;

slint::slint!{
    HelloWorld := Window {
        width: 100px;
        height: 100px;
        Text {
            text: "hello world";
            color: green;
        }
    }
}

fn main() {
    let win = HelloWorld::new();
    println!("width of window is {}", win.get_width());
    win.run();
}
```

Emits following error

```
   Compiling exp_win_size v0.1.0 (C:\tools\msys64\home\dilawar\Work\GITHUB\playground\RUST\slint\exp_win_size)
error[E0599]: no method named `get_width` found for struct `HelloWorld` in the current scope
  --> src\main.rs:16:43
   |
3  | slint::slint!{
   | ------------- method `get_width` not found for this
...
16 |     println!("width of window is {}", win.get_width());
   |                                           ^^^^^^^^^ method not found in `HelloWorld`

```

How do I access the `width` and `height` of `Windows` inside `main`? 

PS: This is a MWE. I have a beautiful GUI almost done using slint. I need to set the `width` of a `SharedPixelBuffer` that is used by `plotter` crate to plot a XY chart. I want to set the width of `SharedPixelBuffer` equal to the width of Window. Hence, I need to get the width of the Window. I have been struggling with this seemingly trivial case.

**Comment by ogoffart** [comment, marked as answer]
Only the property declared in the last element of the .slint are exposed.

The property of the window are actually not exposed.

We would like to extend the native window API so we can query that in native code (see https://github.com/slint-ui/slint/issues/323 and https://github.com/slint-ui/slint/issues/333 )

In the mean time, you can re-export the properties using something like 

```rust
use slint;

slint::slint!{
    HelloWorld := Window {
        width: 100px;
        height: 100px;
        // ###### added these two lines to re-export the size
        property window_width <=> width;
        property window_height <=> height;
        Text {
            text: "hello world";
            color: green;
        }
    }
}

fn main() {
    let win = HelloWorld::new();
    //  ###  used get_window_width()
    println!("width of window is {}", win.get_window_width());
    win.run();
}
```

**Reply by dilawar** [reply]
Thanks @ogoffart. It worked. Also upon resizing the windows, the value is being changed correctly.
